CONTRIBUTING.md

```diff
@@ -119,7 +119,64 @@ ruff .
 
 The license of this project is located in [LICENSE].  By submitting a contribution to this project, you are agreeing that your contribution will be released under the terms of this license.
 
-## Maintainers
+## Roles in Mesa
+Mesa has several roles to help structure our collaboration and recognize great work. They also form a progression path for community members to that want to take on increasing responsibility in the project. Since we're all volunteers, everyone contributes what they can when they can - there are no minimum time commitments. Also, the best ideas and contributions can come from anyone, these roles are just a way to coordinate our efforts.
+
+Feel free to reach out to us anytime to discuss your interests and ambitions in the project. We're always happy to chat about how you can grow your involvement in Mesa!
+
+### Contributor
+Contributors help improve Mesa through:
+- Code contributions
+- Documentation improvements
+- Bug reports and fixes
+- Example models
+- Tutorial improvements
+- Answering questions
+- Participating in discussions
+- Testing pre-releases
+- Sharing Mesa with others
+
+Everyone can contribute what they can, when they can. No contribution is too small! Contributors who have a PR successfully merged receive the "Contributor" label on GitHub.
+
+### Collaborator
+When contributors consistently demonstrate technical skills and community mindset through their contributions, they may be invited to become collaborators. Collaborators help coordinate by:
+
+- Reviewing pull requests
+- Triaging issues and discussions
+- Coordinating between contributors
+- Leading specific areas of development
+- Helping new contributors
+- Participating in project planning
+- Building community
+
+The collaborator role recognizes people who help Mesa grow through both their technical contributions and community involvement. Collaborators receive GitHub triage permissions and the "Collaborator" label.
+
+### Maintainer
+Maintainers help guide Mesa's overall development while ensuring the project remains sustainable and welcoming. They focus on:
+- Project vision and roadmap
+- Major architectural decisions
+- Release management
+- Community governance
+- Mentoring collaborators
+- Final decision-making authority
+- Setting community standards
+- Long-term sustainability
+
+Maintainers are selected based on their technical expertise, project understanding, and community leadership. Maintainers receive full repository permissions and the "Member" label.
+
+### Special Roles
+In some cases, special roles may be created for specific purposes, such as leading particular initiatives or components within Mesa. These roles are created as needed based on project requirements and may come with specific permissions and labels.
```

Comment:
Reading this I immediately think of Mesa Frames, is that the idea? Regardless, I would add an example here just to give folks an idea.

Reply:
Yeah it is indeed. I prefer to keep it a bit open and generic so we don’t have to keep updating it.